content/en/database_monitoring/guide/managed_authentication.md

```diff
@@ -104,7 +147,96 @@ GRANT rds_iam TO datadog;
 4. Complete the Agent setup steps for your [RDS][6] or [Aurora][7] instance.
 
 
-5. [Attach the role][5] with each EC2 instance that is running the agent. Note, this can be done at EC2 launch time.
+{{< tabs >}}
+{{% tab "EC2" %}}
+
+5. Create an IAM role and attach the IAM policy created in step 2 to the role.
+
+```bash
+# Create an IAM role for EC2 instance
+# Replace `<YOUR_IAM_AUTH_DB_ROLE>` with the name of the IAM role
+aws iam create-role --role-name <YOUR_IAM_AUTH_DB_ROLE> --assume-role-policy-document '{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}'
+
+# Attach the IAM policy to the IAM role
+# Replace `<YOUR_IAM_AUTH_DB_POLICY_ARN>` with the ARN of the IAM policy from step 2
+aws iam attach-role-policy --role-name <YOUR_IAM_AUTH_DB_ROLE> --policy-arn <YOUR_IAM_AUTH_DB_POLICY_ARN>
+```
+
+Attach the IAM role to the EC2 instance where the Agent is running. For more information, see [IAM roles for Amazon EC2][1].
+
+[1]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+
+{{% /tab %}}
+{{% tab "ECS Fargate" %}}
+
+5. Create an IAM role and attach the IAM policy created in step 2 to the role.
+
+```bash
+# Create an IAM role for ECS task
+# Replace `<YOUR_IAM_AUTH_DB_ROLE>` with the name of the IAM role
+aws iam create-role --role-name <YOUR_IAM_AUTH_DB_ROLE> --assume-role-policy-document '{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}'
+
+# Attach the IAM policy to the IAM role
+# Replace `<YOUR_IAM_AUTH_DB_POLICY_ARN>` with the ARN of the IAM policy from step 2
+aws iam attach-role-policy --role-name <YOUR_IAM_AUTH_DB_ROLE> --policy-arn <YOUR_IAM_AUTH_DB_POLICY_ARN>
+```
+
+In the ECS task definition, attach the IAM role to the task role where the Agent container is defined. For more information, see [IAM roles for Amazon ECS][1].
+
+[1]: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html
+
+{{% /tab %}}
+{{% tab "EKS" %}}
+
+5. Create an IAM role and attach the IAM policy created in step 2 to the role.
+Map the IAM role to the Kubernetes service account where the Agent is running. For more information, see [IAM roles for Amazon EKS service account][1].
```

Comment:
Let's move this line after the code sample to maintain consistency with the other tabs.